wwwbase/js/scramble.js

```diff
@@ -0,0 +1,1051 @@
+                          if(layers[k].groups[0] == layer.groups[0])
+                          {
+                            downLayers[j] = layers[k];
+                            upLayers[i] = 0;
+                            break;
+                          }
+                        }                    
+                        $("canvas").animateLayerGroup(layer.groups[0],{
+                          x: posX + (j * 65),
+                          y: 200,
+                        }, layerSpeed);
+                        currentWord();
+                        break;
+                      }
+                    }
+                    break;
+                  }
+                }
+              }
+            }
+            else // Move and animate the letter up
+            {
+              for(var i = 0 ; i < upLayers.length; i++) // Check and reposition back into place if draged on the same area
+              {
+                if(upLayers[i] != 0 && upLayers[i].groups[0] == layer.groups[0])
+                {
+                   $("canvas").animateLayerGroup(layer.groups[0],{
+                        x: posX + (i * 65),
+                        y: 50,
+                      }, layerSpeed);
+                   move = false;
+                   break;
+                }
+                else
+                {
+                  move = true;
+                }
+              }
+              if(move)
+              {
+                for(var i = 0 ; i < downLayers.length; i++)
+                {
+                  if(downLayers[i] != 0 && downLayers[i].groups[0] == layer.groups[0])
+                  {
+                    for(var j = 0; j < upLayers.length; j++)
+                    {
+                      if(upLayers[j] == 0)
+                      {
+                        for(var k = 0; k < layers.length; k+=2)
+                        {
+                          if(layers[k].groups[0] == layer.groups[0])
+                          {
+                            upLayers[j] = layers[k];
+                            downLayers[i] = 0;
+                            break;
+                          }                        
+                        } 
+                        $("canvas").animateLayerGroup(layer.groups[0],{
+                          x: posX + (j * 65),
+                          y: 50,
+                        }, layerSpeed);
+                        currentWord();
+                        break;
+                      }                      
+                    }
+                    break;
+                  }                
+                }
+              }
+            }          
+            console.log(upLayers);
+            console.log(downLayers); 
+          },
+        })
+        .animateLayerGroup("boggle" + i, {     
+          x: posX, y: 50
+        }, layerSpeed);
+      }
+    }
+
+   //callAjax(); // init AJAX
+  selectDifficulty();
+  inputListen();
+  //checkWord();
+  testing();
+
+});
```

Comment:
Inca n-am optimizat nimic, or sa dispara ceva linii din zona asta, mai ales switch-urile.
Ideea ca eu am 2 array-uri in care tin minte pozitia obiectelor desenate.
upLayers tine evidenta obiectelor aflate sus si downLayers tine evidenta obiectelor aflate in partea de jos a canvas-ului. (Toate au niste pozitii calculate in functie de index-ul din aceste array-uri)

Atunci cand trag de un element jos, el este sters din upLayers si scris in downLayers, si invers.

Pare ca se repeta mult cod dar trebuie sa tratez doua cazuri. Exista doua layere unul peste altu, layerul de chenar (patratul colorat) si layer-ul de litera. In functie de unde se da click eu trebuie sa fac niste decizii despre unde si cum repozitionez layerele.
Pentru ca pot sa trag de el fie apasand pe chenar fie apasand pe litera. (Am niste console log-uri care indica ce fel de layer a fost tras).
Relatia litera -> chenar o da layer.groups, daca am 4 litere am 4 grupuri de layere, adica in total 8 layere. 
Din pacate e necesar pentru ca informatia despre litera sa afla doar in unul din layere (puteam sa le pun in amandoua dar am zis ca n-avea rost tin aceeasi informatie in 2 locuri separate, pot identifica usor informatia dorita dupa grup in functie de layer-ul selectat).